tests/unit/jobserver/models/test_user.py

```diff
@@ -351,3 +353,20 @@ def test_create_user():
     assert user.email == "test@test.test"
     # Password is not stored in plaintext.
     assert user.password != password
+
+
+def test_user_legacy_roles_round_trip():
```

Comment:
Optionally would like a comment here explaining what we mean by legacy role and why this test exists as it's a little complex.

Reply:
I've added a docstring - do you think that covers it well enough?